### ISSUE TITLE
vmagent adding arbitrary host label when ingesting metrics using datadog ingestion 

### ISSUE THREAD

**Issue by sushovan23** [VictoriaMetrics/VictoriaMetrics#3432]
**Describe the bug**
vmagent version - 1.83.0

When ingesting datadog metrics and not specifying the host explicitly in payload. Then vmagent adds host label to metrics from any of the other metrics which do have host label.

**To Reproduce**
```
echo '
{
  "series": [
    {
        
     
      "metric": "sushovan_system_load_12",
      "interval": 20,
      "points": [
        [
         1669913234,
          12347
        ]
      ],
       "tags": [
         "environment:chase"
       ],
       "type": "rate"
    }
  ]
}
' | curl -X POST --data-binary @- https://vmagent:8429/datadog/api/v1/series

```
Also send some metrics which do have label of host.

When I query the metrics:
```
{
    "status": "success",
    "data": {
        "resultType": "matrix",
        "result": [
            {
                "metric": {
                    "__name__": "sushovan_system_load_12",
                    "environment": "chase",
                    "host": "someIP"
                },
                "values": [
                    [
                        1669913234,
                        "12347"
                    ]
                ]
            },
            {
                "metric": {
                    "__name__": "sushovan_system_load_12",
                    "environment": "chase",
                    "host": "someother_IP"
                },
                "values": [
                    [
                        1669913234,
                        "12347"
                    ]
                ]
            }
        ]
    }
}
```
**Expected behavior**
metric should not have host label from other metrics if not specified.

**Version**
The line returned when passing `--version` command line flag to the binary. For example:
```
vmagent-20221029-004945-tags-v1.83.0-0-g740f7ac5e
```

**Used command-line flags**
xecStart=/app/vmagent/vmagent-prod-v1.83.0 \
    -remoteWrite.rateLimit=${REMOTEWRITERATELIMIT} \
    -remoteWrite.url=${REMOTEWRITEURL} \
    -remoteWrite.maxDiskUsagePerURL=${REMOTEWRITEMAXDISKUSAGEPERURL} \
    -remoteWrite.tmpDataPath=${REMOTEWRITETMPDATAPATH} \
    -promscrape.config=/app/vmagent/scrape_config.yml \
    -tls \
    -tlsCertFile=${TLSCERTFILE} \
    -tlsKeyFile=${TLSKEYFILE}
    
cat /app/vmagent/scrape_config.yml
global:
  scrape_interval: 10s

scrape_configs:
- job_name: vmagent
  scheme: https
  static_configs:
  - targets:
    - localhost:8429

**Comment by zekker6** [comment]
@sushovan23 
VMAgent format expects that host will be passed as a part of series:
```json
{
  "series": [
    {
        
      "host": "some_IP",
      "metric": "sushovan_system_load_12",
      "interval": 20,
      "points": [
        [
         1669913234,
          12347
        ]
      ],
       "tags": [
         "environment:chase"
       ],
       "type": "rate"
    }
  ]
}
```
Could you check if this format helps?

**Comment by valyala** [comment]
The commit 88597f187ba650bd1d4a107bbd266db6ea1eeb76 makes the `host` label optional in [DataDog data ingestion protocol](https://docs.victoriametrics.com/#how-to-send-data-from-datadog-agent).

@sushovan23 , could you build `vmagent` from this commit according to [these instructions](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether it properly processes DataDog data without the `host` label?

**Comment by valyala** [comment]
FYI, `vmagent` and single-node VictoriaMetrics should accept DataDog data ingestion requests without `host` label starting from [v1.85.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @sushovan23 , could you verify whether this fixes the initial issue?

**Comment by valyala** [comment]
It has been appeared that [json.Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal) doesn't reset previous field values (such as `Series.Host`) when the currently parsed message has no these fields. So field values from the previously parsed message may appear in the currently parsed message. This issue should be fixed in the commit 7ae744fce6e2f323280dbed044ae552561ef72e0 .

@sushovan23 , could you check whether the issue is fixed in `vmagent` Docker image with the tag `heads-public-single-node-0-g7ae744fce` (e.g. `docker.io/victoriametrics/vmagent:heads-public-single-node-0-g7ae744fce`)

**Comment by sushovan23** [comment]
@valyala sure i can test it. you want me to test it with v.1.85.0 or the docker image?

**Comment by valyala** [comment]
@sushovan23 , it is better to test the bugfix included in [v1.85.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases), since `v1.85.0` contains incomplete bugfix. If everything works as expected, then, please, close this issue as fixed.

**Comment by valyala** [comment]
The bugfix for this issue has been included in [v1.79.7 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.7) too.

**Comment by valyala** [comment]
This issue has been resolved in the latest releases of vmagent as mentioned above. Closing it then. @sushovan23 , feel free re-opening it or creating new one if it isn't resolved yet.
